cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java

```diff
@@ -63,17 +63,22 @@ public void run(String[] args) throws Exception {
       throw new IllegalArgumentException("Expected to have two arguments: runtime config uri and the runnable name.");
     }
 
+    LocalLocationFactory localLocationFactory = new LocalLocationFactory();
+
     // Localize the runtime config jar
     URI uri = URI.create(args[0]);
-    Location runtimeConfigLocation;
-    if (context.getLocationFactory().getHomeLocation().toURI().getScheme().equals(uri.getScheme())) {
-      runtimeConfigLocation = context.getLocationFactory().create(uri);
+
+    Path runtimeConfigDir;
+    if (localLocationFactory.getHomeLocation().toURI().getScheme().equals(uri.getScheme())) {
+      try (FileInputStream is = new FileInputStream(new File(uri))) {
+        runtimeConfigDir = expand(uri, is, Paths.get(Constants.Files.RUNTIME_CONFIG_JAR));
+      }
     } else {
-      runtimeConfigLocation = new LocalLocationFactory().create(new File(uri).toURI());
+      try (InputStream  is = context.openHttpURLConnection(fileDownloadURLPath(uri)).getInputStream()) {
```

Comment:
Remove extra space before `is`

Reply:
You gotta disconnect the URLConnection as well

Reply:
Removed the extra space.

For disconnect, intentionally omitted it so underlying socket can be reused.
https://stackoverflow.com/a/11533423

But if you think we should always disconnect, can change to that.

Reply:
The stackoverflow is a bit misleading. Since we close the InputStream here, calling disconnect will allow the connection to be reused (1st point in the SO).

Calling disconnect is for the case that if the response is non-200/300, and we don't read/close the Error stream, calling disconnect will make sure the connection won't get reused and the socket get closed correct (without leaking resources). Non connection reused is what we wanted in this case.

Alternatively, if you want to reused a socket after the response is non-200/300, you have to read and close the Error stream (`getErrorStream`).

Reply:
Changed to disconnecting upon closing stream. PTAL